src/api/index.ts

```diff
@@ -0,0 +1,23 @@
+import { getApiUri } from '../misc/url'
+import { Snippet, RawSnippet } from '../store'
+
+export const fetchSnippet = (id: number) => {
+  return fetch(getApiUri(`snippets/${id}`))
+    .then(response => response.json())
+}
+
+export const postSnippet = (snippet: RawSnippet, onSuccess: (snippet: Snippet) => void, onError = () => {}) => {
+  return fetch(getApiUri('snippets'), {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(snippet),
+  })
+    .then(response => response.json())
+    .then((json) => {
```

Comment:
minor, but there is no need to create an anonymous function `.then(onSuccess)` should work